src/plugins/autofill/autofill.js

```diff
@@ -231,47 +234,80 @@ export class Autofill extends BasePlugin {
 
     if (startOfDragCoords && startOfDragCoords.row > -1 && startOfDragCoords.col > -1) {
       const selectionData = this.getSelectionData();
-      const beforeAutofillHook = this.hot.runHooks('beforeAutofill', startOfDragCoords, endOfDragCoords, selectionData);
 
-      if (beforeAutofillHook === false) {
+      const sourceRange = {
+        from: {
+          row: Math.min(selectionRangeLast.from.row, selectionRangeLast.to.row),
+          col: Math.min(selectionRangeLast.from.col, selectionRangeLast.to.col)
+        },
+        to: {
+          row: Math.max(selectionRangeLast.from.row, selectionRangeLast.to.row),
+          col: Math.max(selectionRangeLast.from.col, selectionRangeLast.to.col)
+        }
+      };
+
+      const targetRange = {
+        from: {
+          row: Math.min(startOfDragCoords.row, endOfDragCoords.row),
+          col: Math.min(startOfDragCoords.col, endOfDragCoords.col)
+        },
+        to: {
+          row: Math.max(startOfDragCoords.row, endOfDragCoords.row),
+          col: Math.max(startOfDragCoords.col, endOfDragCoords.col)
+        }
+      };
```

Comment:
```suggestion
      const targetRange = new CellRange(startOfDragCoords, startOfDragCoords, endOfDragCoords);
```

Reply:
To be honest I didn’t use CellRange specifically for the reason that it had the `highlight` property, is this really a good idea? Including a redundant `highlight` in all of the hooks could be confusing for users, no?

Reply:
I've never seen that someone have some confusion with `highlight`. Most users are focused on `from` and `to` properties. I can agree that in this case, the `highlight` is redundant. But if I had chosen I would go with my suggestion. The `CellRange` (with `CellCoords`) class is used everywhere where the ranges need to be stored. You can still meet in a few places where the ranges are created using literal objects but this should be treated as a legacy solution. 

I added the suggestion because in the past I fixed some issues related to custom coords objects. Generally, they are not synced with underneath ranges. They include (or not) negative values, the coords are constructed in a way that works only when someone selected the cells from the left to right, in opposite there was an error, etc.

Reply:
There you go, https://github.com/handsontable/handsontable/pull/7648/commits/620d7347daceef1b5db4881272286037ce0fc58f, I'll need a solution for these though:
https://github.com/handsontable/handsontable/commit/620d7347daceef1b5db4881272286037ce0fc58f#diff-19f550dd8ca624aa45c467d1c40310ff49d805005539e416be5aaa9400218632R364-R369
https://github.com/handsontable/handsontable/commit/620d7347daceef1b5db4881272286037ce0fc58f#diff-19f550dd8ca624aa45c467d1c40310ff49d805005539e416be5aaa9400218632R524-R529